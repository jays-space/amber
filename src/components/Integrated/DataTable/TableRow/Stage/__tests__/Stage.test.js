import { fireEvent, render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Stage } from "..";

describe("<Stage />", () => {
  const currentStage = [1, 2, 3, 4, 5];

  it("should render correctly", () => {
    const select = true;
    render(<Stage currentStage={currentStage[2]} select={select} />);

    const stageComponent = screen.getByTitle("application-stage");
    expect(stageComponent).toBeInTheDocument();
    expect(stageComponent).toMatchSnapshot();
  });

  it("should simulate stage change", () => {
    const select = true;
    render(<Stage currentStage={currentStage[2]} select={select} />);

    const selectElement = screen.getByTestId("stage-selector");

    fireEvent.change(selectElement, { target: { value: currentStage[3] } });
    let stages = screen.getAllByTestId("stage-option");
    expect(stages[0].selected).toBeFalsy();
    expect(stages[1].selected).toBeFalsy();
    expect(stages[2].selected).toBeFalsy();
    expect(stages[3].selected).toBeTruthy();
    expect(stages[4].selected).toBeFalsy();
  });

  for (let stage of currentStage) {
    it(`should render the stage ${stage}`, () => {
      const select = true;
      render(<Stage currentStage={stage} select={select} />);

      const stageComponent = screen.getByTitle("application-stage");
      expect(stageComponent).toBeInTheDocument();
    });
  }
});
