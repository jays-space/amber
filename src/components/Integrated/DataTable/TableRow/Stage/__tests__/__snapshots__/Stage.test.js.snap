// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Stage /> should render correctly 1`] = `
<div
  class="relative z-20"
  title="application-stage"
>
  <select
    class="w-full lg:w-fit text-lg font-semibold text-slate-700 capitalize after:absolute after:top-3 after:right-3 after:w-0 after:h-0 after:border-none focus-visible:outline-none"
    data-testid="stage-selector"
    name="stage"
  >
    <option
      class="p-10 text-lg"
      data-testid="stage-option"
      value="1"
    >
      new applied
    </option>
    <option
      class="p-10 text-lg"
      data-testid="stage-option"
      value="2"
    >
      screening
    </option>
    <option
      class="p-10 text-lg"
      data-testid="stage-option"
      value="3"
    >
      code challenge
    </option>
    <option
      class="p-10 text-lg"
      data-testid="stage-option"
      value="4"
    >
      interview
    </option>
    <option
      class="p-10 text-lg"
      data-testid="stage-option"
      value="5"
    >
      hire
    </option>
  </select>
  <div
    class=" mt-2 flex"
  >
    <div
      class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-orange-500"
    >
      <p
        class="tracking-wide  text-slate-700 text-white"
        data-testid="paragraph"
      >
        1
      </p>
    </div>
    <div
      class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-orange-500"
    >
      <p
        class="tracking-wide  text-slate-700 text-white"
        data-testid="paragraph"
      >
        2
      </p>
    </div>
    <div
      class="px-2.5 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-orange-500"
    >
      <p
        class="tracking-wide  text-slate-700 text-white"
        data-testid="paragraph"
      >
        3
      </p>
    </div>
    <div
      class="px-3 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-neutral-300"
    />
    <div
      class="px-3 mx-0.5 first-of-type:ml-0 flex justify-center items-center rounded bg-neutral-300"
    />
  </div>
</div>
`;
